net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+#   for each install or upgrade
+# * Keep personal data **outside** of ${prefix}, but provide symlinks
+
+destroot {
+    # configuration design: MacPorts file and/or directory templates installed
+    # to *.macports, then edited with local network settings, then in
+    # post-activate copied to actual configuration files if such don't exist 
+
+    xinstall -m 0755 -o ${calendarserverUser} -d \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/${ccsname}
+    destroot.keepdirs-append \
+        ${destroot}${calendarserverdir} \
+        ${destroot}${calendarserverdir}/Library/CalendarServer \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/${ccsname}
+    foreach d {
+        Library/CalendarServer/Config
+        Library/CalendarServer/etc
+        Library/CalendarServer/etc/nginx_root
+        Library/CalendarServer/logs
+        Library/CalendarServer/postgresql
+        Library/CalendarServer/run
+        } {
+        xinstall -m 0750 -o ${calendarserverUser} -g _calendar -d \
+            ${destroot}${calendarserverdir}/${d}
+        destroot.keepdirs-append \
+            ${destroot}${calendarserverdir}/${d}
+    }
+    file attributes \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx_root \
+        -permissions 0755
+    xinstall -m 0755 -d \
+        ${destroot}${prefix}/src/${ccsname}
+    # calendarserver runtime sockets
+    foreach d {
+        caldavd
+        caldavd_requests
+        } {
+        xinstall -m 0770 -o ${calendarserverUser} -g _calendar -d \
+            ${destroot}${calendarserverdir}/Library/CalendarServer/run/${d}
+        destroot.keepdirs-append \
+            ${destroot}${calendarserverdir}/Library/CalendarServer/run/${d}
+    }
+
+    # symlink to personal data outside of ${prefix}
+    ln -s /private/var/${calendarserverUser}/Library/CalendarServer/Data \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/Data
+
+    # source
+    move ${worksrcpath} \
+        ${destroot}${prefix}/src/${ccsname}
+
+    # calenderserver configuration
+    xinstall -m 0600 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/calendarserver.plist \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/Config/calendarserver.plist.macports
+
+    # nginx reverse proxy
+    xinstall -m 0640 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/nginx.conf \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx.conf.macports
+    xinstall -m 0644 -o ${calendarserverUser} -g _calendar \
+        ${filespath}/default.html \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/etc/nginx_root
+
+    # ccs-calendarserver's postgres code points to `PSQL = "../postgresql/_root/bin/psql"`
+    ln -s ${prefix} \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/postgresql/_root
+}
+
+# Network configuration
+# hard-coded examples
+set host                host
+set domain              domain
```

Comment:
I don't understand how this works. Is the user supposed to modify the `Portfile` to get this working as desired?

Reply:
These are hard-coded defaults so the Portfile doesn’t have to make system calls to determine network settings.

The `post-activate` block gets actual network settings during install and sets up configuration files appropriately.

Users will have.to test and review all settings prior to deployment, as noted in `notes`.